### PR TITLE
Improvement of Timetable DB and Data classes

### PR DESCRIPTION
**Pull request by koji-1009**
## Issue
- ref #628 #629

## Overview (Required)
- JSON parsing class has been extended to match JSON.
- JSON can now be parsed and cached in the DB.

## Links
- none

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by takahirom** [comment]
Are you an AI? 😄  Your implementation speed is too fast. 🤖

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/DroidKaigi2021Api.kt [hunk 1]
```diff
@@ -433,33 +439,61 @@ internal fun SessionAllResponse.toTimetableContents(): TimetableContents {
                 TimetableSpeaker(apiSpeaker.fullName!!, apiSpeaker.profilePicture)
             }.first()
         }
+    val categoryIdToCategory: Map<Int, TimetableCategory> = feedContents.categories!!
+        .flatMap { it.items!! }
```
Comment (koji-1009) [inline data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/DroidKaigi2021Api.kt:443]: It is also possible to reference only "Category" in categories, in which case we can use the order or id of "Category". 
The current implementation targets the items contained in all categories.

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/session/response/SessionResponse.kt [hunk 2]
```diff
@@ -19,7 +19,6 @@ internal data class SessionResponse(
     val isPlenumSession: Boolean,
     val targetAudience: String,
     val interpretationTarget: Boolean,
-    val videoUrl: String? = null,
-    val slideUrl: String? = null,
+    val asset: SessionAssetResponse,
```
Comment (koji-1009) [inline data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/session/response/SessionResponse.kt:22]: Fixed to match JSON.

File: data/db/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableItemDao.kt [hunk 3]
```diff
@@ -73,6 +78,18 @@ private fun TimetableItemSessionQueries.insert(session: TimetableItem.Session) {
...
             endsAt = session.endsAt.toEpochMilliseconds(),
+            jaCategory = session.category.title.jaTitle,
+            enCategory = session.category.title.enTitle,
+            targetAudience = session.targetAudience,
+            language = session.language,
+            assetSlideUrl = session.asset.slideUrl,
+            assetVideoUrl = session.asset.videoUrl,
+            jaRoomName = session.room.name.jaTitle,
+            enRoomName = session.room.name.enTitle,
+            levels = session.levels.joinToString(stringListDivider),
```
Comment (koji-1009) [inline data/db/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableItemDao.kt:89]: https://cashapp.github.io/sqldelight/jvm_sqlite/types/

We can use `List` in `.sq` file, but we must add `ColumnAdapter<List<String>, String>` object to every Database Instance.
So, use `joinToString` and `split` method.

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/DroidKaigi2021Api.kt (hunk 1 above)
Comment (koji-1009) [reply data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/DroidKaigi2021Api.kt:443]: Note: In the current JSON, "sessionCategoryItemId" is 89435, which specifies the "Session format".

**Comment by koji-1009** [comment]
Oops, I'll rebase this.

File: model/src/commonMain/kotlin/io/github/droidkaigi/feeder/TimetableItem.kt [hunk 4]
```diff
@@ -9,20 +9,62 @@ sealed class TimetableItem(
...
     data class Special(
         override val id: String,
         override val title: MultiLangText,
         override val startsAt: Instant,
         override val endsAt: Instant,
+        override val category: TimetableCategory,
+        override val targetAudience: String,
+        override val language: String,
+        override val asset: TimetableAsset,
+        override val room: TimetableRoom,
```
Comment (takahirom) [inline model/src/commonMain/kotlin/io/github/droidkaigi/feeder/TimetableItem.kt:55]: There will be no parallel sessions at DroidKaigi 2021. Therefore, no room is needed. 🙇

File: data/db/src/commonMain/sqldelight/io/github/droidkaigi/feeder/data/TimetableItemSession.sq [hunk 5]
```diff
@@ -3,7 +3,19 @@ CREATE TABLE timetableItemSession (
     jaTitle TEXT NOT NULL,
     enTitle TEXT NOT NULL,
     startsAt INTEGER NOT NULL,
-    endsAt INTEGER NOT NULL
+    endsAt INTEGER NOT NULL,
```
Comment (takahirom) [inline data/db/src/commonMain/sqldelight/io/github/droidkaigi/feeder/data/TimetableItemSession.sq:6]: Can you add `3.sqm` empty file in src/commonMain/sqldelight/io/github/droidkaigi/feeder/data like 2.sqm?
It is needed for the migration 🙏

File: data/db/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableItemDao.kt [hunk 6]
```diff
@@ -65,6 +68,8 @@ internal class TimetableItemDaoImpl(database: Database) : TimetableItemDao {
     }
 }
 
+private val stringListDivider = ","
```
Comment (takahirom) [inline data/db/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableItemDao.kt:71]: You can use `const` 🔹

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/DroidKaigi2021Api.kt (hunk 1 above)
Comment (takahirom) [reply data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/DroidKaigi2021Api.kt:443]: I checked too. I think it's okay. 🙆‍♀️

**Review by takahirom — COMMENTED** [review]
LGTM! 🏎️  Can you check some nits? 🙏

File: data/db/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableItemDao.kt (hunk 6 above)
Comment (koji-1009) [reply data/db/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/TimetableItemDao.kt:71]: My bad!

File: data/db/src/commonMain/sqldelight/io/github/droidkaigi/feeder/data/TimetableItemSession.sq (hunk 5 above)
Comment (koji-1009) [reply data/db/src/commonMain/sqldelight/io/github/droidkaigi/feeder/data/TimetableItemSession.sq:6]: I added the file. 1293754

File: model/src/commonMain/kotlin/io/github/droidkaigi/feeder/TimetableItem.kt (hunk 4 above)
Comment (koji-1009) [reply model/src/commonMain/kotlin/io/github/droidkaigi/feeder/TimetableItem.kt]: Oh, I see.
I deleted TimetableRoom class.

**Review by takahirom — APPROVED** [review]
Thank you for creating the foundation for DroidKaigi 2021!

**Pull request merged**